#56: Added Slick Unit Test

satendrakumar opened the pull request.

satendrakumar committed:
#56: Added Slick Unit Test

satendrakumar committed:
#56: Added Slick Unit Test

lightbend-cla-validator commented:
Hi @satendrakumar,

Thank you for your contribution! We really value the time you've taken to put this together.

Before we proceed with reviewing this pull request, please sign the Lightbend Contributors License Agreement:

[http://www.lightbend.com/contribute/cla](http://www.lightbend.com/contribute/cla)

satendrakumar commented:
I signed  the CLA

satendrakumar committed:
Update unit test using ScalaTest

satendrakumar committed:
Update unit test using ScalaTest

satendrakumar committed:
Update gradle build

satendrakumar committed:
Update gradle build

satendrakumar committed:
Remove JUnitRunner

satendrakumar committed:
Remove JUnitRunner

satendrakumar commented:
@wsargent Done.

marcospereira committed:
Updated with template-control on 2018-07-16T18:38:45.495Z (#70)

**/plugins.sbt:
    addSbtPlugin("com.typesafe.play" % "sbt-plugin" % "2.6.16")
  **build.gradle:
    def playVersion = "2.6.16"

marcospereira committed:
Updated with template-control on 2018-07-19T01:58:55.997Z (#71)

**/plugins.sbt:
    addSbtPlugin("com.typesafe.play" % "sbt-plugin" % "2.6.17")
  **build.gradle:
    def playVersion = "2.6.17"

marcospereira committed:
Updated with template-control on 2018-08-20T20:37:13.878Z (#73)

**/build.properties:
    sbt.version=1.2.1
  **/plugins.sbt:
    addSbtPlugin("com.typesafe.play" % "sbt-plugin" % "2.6.18")
  **build.gradle:
    def playVersion = "2.6.18"
  **gradle/wrapper/gradle-wrapper.properties:
    distributionUrl=https\://services.gradle.org/distributions/gradle-4.9-bin.zip

marcospereira committed:
Updated with template-control on 2018-09-11T20:14:50.417Z (#74)

**/plugins.sbt:
    addSbtPlugin("com.typesafe.play" % "sbt-plugin" % "2.6.19")
  **build.gradle:
    def playVersion = "2.6.19"

octonato committed:
Upgrade branch 2.6.x using TemplateControl (#75)

```
Updated with template-control on 2018-10-08T19:11:41.281Z
  **/plugins.sbt:
    addSbtPlugin("com.typesafe.play" % "sbt-plugin" % "2.6.20")
  **build.gradle:
    def playVersion = "2.6.20"

```

octonato committed:
Upgrade branch 2.6.x using TemplateControl (#78)

* Updated with template-control on 2018-11-29T15:50:43.767Z
  /.mergify.yml:
    wrote /.mergify.yml
  **build.sbt:
    scalaVersion := "2.12.7"

* Remove JDK 9 & 10

marcospereira committed:
Updated with template-control on 2019-01-04T17:13:23.925Z (#79)

/.travis.yml:
    wrote /.travis.yml
  **build.sbt:
    scalaVersion := "2.12.8"
  **/build.properties:
    sbt.version=1.2.8

octonato committed:
Updated with template-control on 2019-01-08T14:44:41.666Z (#85)

**/plugins.sbt:
    addSbtPlugin("com.typesafe.play" % "sbt-plugin" % "2.6.21")
  **build.gradle:
    def playVersion = "2.6.21"

octonato committed:
Upgrade branch 2.6.x using TemplateControl (#88)

```
Updated with template-control on 2019-01-16T12:08:45.037Z
  /LICENSE:
    wrote /LICENSE
  /NOTICE:
    wrote /NOTICE
  /.mergify.yml:
    wrote /.mergify.yml

```

octonato committed:
Updated with template-control on 2019-01-17T15:10:58.253Z (#89)

/.mergify.yml:
    wrote /.mergify.yml

octonato committed:
Upgrade branch 2.6.x using TemplateControl (#90)

```
Updated with template-control on 2019-02-01T10:40:33.775Z
  /.mergify.yml:
    wrote /.mergify.yml

```

octonato commented:
This PR was originally sent to 2.6.x, but default branch is now 2.7.x. 

We should cherrypick it on 2.7.x once merged.

octonato commented:
CLA signed, but check didn't capture it.
https://www.lightbend.com/contribute/cla/check/satendrakumar

satendrakumar commented:
@renatocaval Is something missing from my end?

octonato commented:
@satendrakumar, no it's not. We just didn't have yet the opportunity to review and merge it. :-)

Thanks a lot for your contribution.

marcospereira commented on conf/test.conf:
```diff
@@ -0,0 +1,5 @@
+include "application.conf"
```
Does not looks necessary since it is using basically the same values in `application.conf`.

marcospereira replied:
Or is this here to demonstrate how to have a distinct configuration for tests? If so, then I think it could be a good idea to also add a comment explaining it.

satendrakumar replied:
@marcospereira The reason for including an application.conf some configurations are common for all environments.

Generally application.conf we used for dev environment. For running unit test we have to change some configurations(for eg:  database credentials ) and some reuse from the application.conf.

octonato replied:
@satendrakumar, yep, we all agree that's a good thing to do. I think Marcos point is that if we are using that sample to also educate people on how to tweak configurations for different environments, then it's better to leave a small comment explaining the technique.

satendrakumar replied:
@renatocaval thanks for the explanation. Done.

marcospereira commented on build.sbt:
```diff
@@ -11,7 +11,10 @@ crossScalaVersions := Seq("2.11.12", "2.12.4")
 
 libraryDependencies += specs2 % Test
+
+javaOptions in Test += "-Dconfig.file=conf/test.conf"
```
If `test.conf` is not necessary, we must also remove this line.

satendrakumar replied:
@marcospereira yes, You are right. It is not necessary. 
For new users, it is easy to have this. They will keep all test configuration in test.conf. If they are using `sbt test ` all the configuration will come from test.conf.

marcospereira reviewed (commented):
Looking good to me. Thank you, @satendrakumar.

marcospereira commented:
Closing and re-opening to see if the CLA bot will get it.

marcospereira committed:
Updated with template-control on 2019-02-13T20:25:39.652Z (#92)

/.mergify.yml:
    wrote /.mergify.yml

satendrakumar committed:
Merge branch 'issue#56' of github.com:satendrakumar/play-scala-slick-example into issue#56

satendrakumar committed:
Update test configuration file
